### PR TITLE
Added `aws-adapter` Helm chart

### PR DESCRIPTION
**Pull request by zeborg**
Signed-off-by: Abhinav Sinha <abhinav@nirmata.com>

cc - @patelrit @pns-nirmata @prateekpandey14

### CODE REVIEW

File: charts/aws-adapter/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
+
+{{/*
+Generate the dockerconfigjson value
+*/}}
+{{- define "nirmata-aws-adapter.dockerconfigjson" -}}
+{{- $user_pwd_hashed := printf "%s:%s" .Values.registryConfig.username .Values.registryConfig.password | b64enc }}
+{{- printf "{\"auths\":{\"ghcr.io\":{\"auth\":\"%s\"}}}" $user_pwd_hashed | b64enc }}
+{{- end }}
```
Comment (pns-nirmata) [inline charts/aws-adapter/templates/_helpers.tpl:59]: Github flags an error about no newline at end of file. Needs single newline at the end.

File: charts/aws-adapter/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        # More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+        resources:
+          limits:
+            cpu: 500m
+            memory: 128Mi
+          requests:
+            cpu: 10m
+            memory: 64Mi
+      serviceAccountName: nirmata-aws-adapter-sa
+      terminationGracePeriodSeconds: 10
```
Comment (pns-nirmata) [inline charts/aws-adapter/templates/deployment.yaml:69]: Mostly, the deployment will fail some policy checks like the PSS restricted profile. Or even the best practices and other kyverno standard policy sets. Need to add those fields. E.g. see https://github.com/nirmata/imagekey-controller/commit/ac6d87e287e5157bade549b9ecbb4a75cba9306a#diff-ce0d597a9e5f20a8fe61c3ba5e6185a6b90cfef2c00bb5b5ef3e97b15b79f480

Comment (zeborg) [reply charts/aws-adapter/templates/deployment.yaml:69]: @pns-nirmata - do you mean these?
https://github.com/nirmata/kyverno-charts/pull/29/files#diff-e2c2d2700e0fb4fdec6d2b4b25e08a4f6956e10d2ae113ad5cb860e6c9df0c20R37-R46

File: charts/aws-adapter/templates/_helpers.tpl (hunk 1 above)
Comment (zeborg) [reply charts/aws-adapter/templates/_helpers.tpl]: Fixed 🫡

File: charts/aws-adapter/templates/deployment.yaml (hunk 2 above)
Comment (pns-nirmata) [reply charts/aws-adapter/templates/deployment.yaml:69]: Yes. That one. Could be others. Run a `trivy config .` Additionally, just run the pod. Install default policies (best practices, security, multitenancy) in Kyverno and check for other violations.

File: charts/aws-adapter/crds/security.nirmata.io_awsconfigs.yaml [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+                format: date-time
+                type: string
+            required:
+            - lastPollInfo
+            type: object
+        type: object
+    served: true
+    storage: true
+    subresources:
+      status: {}
```
Comment (pns-nirmata) [inline charts/aws-adapter/crds/security.nirmata.io_awsconfigs.yaml:326]: Also, will be good to have comments for various CRD fields for someone going through it. Adding comments in the struct in *_types.go should achieve it.

Comment (zeborg) [reply charts/aws-adapter/crds/security.nirmata.io_awsconfigs.yaml:326]: Will do that, thanks!

Comment (prateekpandey14) [reply charts/aws-adapter/crds/security.nirmata.io_awsconfigs.yaml:326]: Are we fixing the CRDs comment in this PR ?

Comment (zeborg) [reply charts/aws-adapter/crds/security.nirmata.io_awsconfigs.yaml:326]: Hey @prateekpandey14, I'll be working on all the fixes including the vulnerability issue as separately mentioned by Ritesh for AWS adapter today. Sorry for the delay.

File: charts/aws-adapter/values.yaml [hunk 4]
```diff
@@ -30,7 +30,7 @@ image:
   repository: ghcr.io/nirmata/aws-adapter
   pullPolicy: Always
   # Overrides the image tag whose default is the chart appVersion.
-  tag: v0.1.0
+  tag: latest
```
Comment (prateekpandey14) [inline charts/aws-adapter/values.yaml:33]: we should not use the latest as tag , instead the version should be v0.1.1 ?

Comment (zeborg) [reply charts/aws-adapter/values.yaml]: Fixed 🫡

**Review by prateekpandey14 — APPROVED** [review]

**Pull request merged**